cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -156,6 +156,22 @@ func (g *gitHubRepository) ComponentsPath() string {
 
 // GetFile returns a file for a given provider version.
 func (g *gitHubRepository) GetFile(version, path string) ([]byte, error) {
+	cacheID := fmt.Sprintf("%s/%s:%s:%s", g.owner, g.repository, version, path)
+	if content, ok := cacheFiles[cacheID]; ok {
+		return content, nil
+	}
+
+	// Try to get the file using http get.
+	// NOTE: this can be disabled by setting GORPOXY to `direct` or `off` (same knobs used for skipping goproxy requests).
+	if goProxyClient, _ := g.getGoproxyClient(); goProxyClient != nil {
+		if files, err := g.httpGetFilesFromRelease(version, path); err == nil {
+			cacheFiles[cacheID] = files
+			return files, nil
+		}
```

Comment:
Should the error get logged to notify that we fallback to the github api?

Reply:
I have added a log line in case of error, similar to what is done when we fail to get releases via goproxy